File: tracker/src/main/java/org/matomo/sdk/dispatcher/DefaultDispatcherFactory.java
package org.matomo.sdk.dispatcher;

import org.matomo.sdk.Tracker;
import org.matomo.sdk.tools.Connectivity;

public class DefaultDispatcherFactory implements DispatcherFactory {
    public Dispatcher build(Tracker tracker) {
        PacketSender packetSender;
        if (tracker.getOkHttpClient()!= null) {
            packetSender = new DefaultPacketSender();
        } else {
            packetSender = new OkHttpPacketSender(tracker.getOkHttpClient());
        }

        return new DefaultDispatcher(
                new EventCache(new EventDiskCache(tracker)),
                new Connectivity(tracker.getMatomo().getContext()),
                new PacketFactory(tracker.getAPIUrl()),
                packetSender
        );
    }
}

File: tracker/src/main/java/org/matomo/sdk/dispatcher/DefaultDispatcherFactory.kt
package org.matomo.sdk.dispatcher

import org.matomo.sdk.Tracker
import org.matomo.sdk.tools.Connectivity

open class DefaultDispatcherFactory : DispatcherFactory {
    override fun build(tracker: Tracker): Dispatcher {
        return DefaultDispatcher(
            EventCache(EventDiskCache(tracker)),
            Connectivity(tracker.matomo.context),
            PacketFactory(tracker.apiUrl),
            DefaultPacketSender(), {}
        )
    }
}
